File: 234-palindrome-linked-list/234-palindrome-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


bool isPalindrome(struct ListNode* head) {
    // copy the list into an array
    int array[100000];
    struct ListNode *curr = head;
    int count = 0;
    while (curr != NULL) {
        array[count] = curr->val;
        count++;
        curr = curr->next;
    }
    
    // check palindrome 
    for (int i = 0; i < count/2; i++) {
        if (array[i] != array[count-1-i]) {
            return false; 
        }
    }
    return true;

}

File: 234-palindrome-linked-list/234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        
        # copy the values into a list 
        values = []
        while head is not None:
            values.append(head.val)
            head = head.next
        
        reversed_values = values[::-1]
        
        if reversed_values == values:
            return True
        else:
            return False
            
        